src/shared/particle_dynamics/execution_unit/execution_proxy.hpp

```diff
@@ -0,0 +1,54 @@
+#ifndef SPHINXSYS_EXECUTION_PROXY_HPP
+#define SPHINXSYS_EXECUTION_PROXY_HPP
+
+#include "execution_policy.h"
+#include "execution_context.hpp"
+
+namespace SPH {
+    namespace execution {
+        template<typename BaseT, typename KernelT>
+        class ExecutionProxy {
+        public:
+            using Base = BaseT;
+            using Kernel = KernelT;
+
+            ExecutionProxy(BaseT* base, KernelT* proxy) : base(base), kernel(proxy) {}
+
+            template<class ExecutionPolicy = ParallelPolicy>
+            BaseT* get(const ExecutionPolicy& = par,
+                       std::enable_if_t<std::negation_v<std::is_same<ExecutionPolicy, ParallelSYCLDevicePolicy>>>* = nullptr) const {
+                return base;
+            }
```

Comment:
It can be simplified.
```suggestion
            template<class ExecutionPolicy = ParallelPolicy>
            BaseT* get(const ExecutionPolicy& = par) const { return base; }
```